parquet-hadoop/src/main/java/org/apache/parquet/hadoop/rewrite/ParquetRewriter.java

```diff
@@ -324,15 +324,20 @@ private void processBlocksFromReader(IndexCache indexCache) throws IOException {
 
           // Translate compression and/or encryption
           writer.startColumn(descriptor, crStore.getColumnReader(descriptor).getTotalValueCount(), newCodecName);
-          processChunk(
+          boolean needOverwriteStatistics = processChunk(
                   chunk,
                   newCodecName,
                   columnChunkEncryptorRunTime,
                   encryptColumn,
                   indexCache.getBloomFilter(chunk),
                   indexCache.getColumnIndex(chunk),
                   indexCache.getOffsetIndex(chunk));
-          writer.endColumn();
+          if (needOverwriteStatistics) {
+            // All the column statistics are invalid, so we need to overwrite the column statistics
+            writer.endColumn(chunk.getStatistics());
```

Comment:
Is it cleaner to handle this inside processChunk()? Then we don't have to bother with endColumn()

Reply:
I think that would be better. And just need to call the API: `public void invalidateStatistics(Statistics<?> totalStatistics)` in the `processChunk`.

Reply:
Updated